code/obj/item/cable_coil.dm

```diff
@@ -234,182 +225,65 @@ obj/item/cable_coil/abilities = list(/obj/ability_button/cable_toggle)
 
 /obj/item/cable_coil/attackby(obj/item/W, mob/user)
 	if (issnippingtool(W) && src.amount > 1)
-		src.amount--
-		take(1, user.loc)
-		boutput(user, "You cut a piece off the [base_name].")
-		src.UpdateIcon()
+		var/obj/item/cable_coil/A = split_stack(round(input("How long of a wire do you wish to cut?","Length of [src.amount]",1) as num))
+		if (istype(A))
+			A.set_loc(user.loc) //Hey, split_stack, Why is the default location for the new item src.loc which is *very likely* to be a damn mob?
```

Comment:
wouldn't we want get turf?

Reply:
I don't think it matters that much, but sure. It's a little weird if you were to cut cable while in a locker or something but frankly I doubt things like that occur in practice